src/shared/components/table/components/main/components/table-row/TableRow.tsx

```diff
@@ -0,0 +1,123 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import classNames from 'classnames';
+import React, { useContext } from 'react';
+
+import type { TableRowData } from 'src/shared/components/table/state/tableReducer';
+import { TableContext } from 'src/shared/components/table/Table';
+
+import TableCell from '../table-cell/TableCell';
+import RowSelector from './components/row-selector/RowSelector';
+
+import styles from 'src/shared/components/table/Table.scss';
+
+const TableRow = (props: { rowData: TableRowData; rowId: string }) => {
+  const {
+    isSelectable,
+    searchText,
+    dispatch,
+    columns,
+    hiddenColumnIds,
+    hiddenRowIds,
+    data
+  } = useContext(TableContext) || {
+    isSelectable: true
+  };
+
+  if (!(data && props.rowData && dispatch && columns && hiddenRowIds)) {
+    return null;
+  }
+
+  if (hiddenRowIds[props.rowId]) {
+    return null;
+  }
```

Comment:
How about putting this check in the previous if condition since both return `null`?